src/main/java/lotto/domain/Lottos.java

```diff
@@ -20,12 +51,44 @@ private void generateLottoGroup(int count) {
         }
     }
 
-    public static String findResult(Map<Rank, Integer> countByRank) {
-        Money.findEarning(countByRank);
-        return Money.findEarningRate();
+    public ArrayList<Rank> drawLotto(WinningLotto winningLotto) {
+        ArrayList<Rank> wins = new ArrayList<>();
+        for (Lotto lotto : lottoGroup) {
+            Rank rank = winningLotto.findRank(lotto);
+            wins.add(rank);
+        }
+        return wins;
```

Comment:
```suggestion
        return new Result(wins);
```

```
public class Result {
    private Map<Rank, Integer> countByRank;

    public Result(List<Rank> wins) {
        Map<Rank, Integer> countByRank = new TreeMap<>();
        for (int i = 1; i < Rank.values().length; i++) {
            Rank rank = Rank.values()[i];
            int rankCount = (int) wins.stream().filter(win -> win == rank).count();
            countByRank.put(rank, rankCount);
        }
    }
}
```

이런 Result객체를 만들어서 Controller에 있는 countEachRank로직을 이동시켜도 좋을 것 같네요.